worlds/stardew_valley/regions.py

```diff
@@ -1,312 +1,611 @@
+        if not unreachable_regions:
+            return randomized_connections
+        swap_one_connection(regions_by_name, connections_by_name, randomized_connections, reachable_regions,
+                            unreachable_regions, connections_to_randomize, random)
+
+
+def find_reachable_regions(regions_by_name, connections_by_name,
+                           randomized_connections: Dict[ConnectionData, ConnectionData]):
+    reachable_regions = {Region.menu}
+    unreachable_regions = {region for region in regions_by_name.keys()}
+    unreachable_regions.remove(Region.menu)
+    exits_to_explore = list(regions_by_name[Region.menu].exits)
+    while exits_to_explore:
+        exit_name = exits_to_explore.pop()
+        exit_connection = connections_by_name[exit_name]
+        replaced_connection = randomized_connections[exit_connection]
+        target_region_name = replaced_connection.destination
+        if target_region_name in reachable_regions:
+            continue
+
+        target_region = regions_by_name[target_region_name]
+        reachable_regions.add(target_region_name)
+        unreachable_regions.remove(target_region_name)
+        exits_to_explore.extend(target_region.exits)
+    return reachable_regions, unreachable_regions
+
+
+def swap_one_connection(regions_by_name, connections_by_name,randomized_connections: Dict[ConnectionData, ConnectionData],
+                        reachable_regions: Set[str], unreachable_regions: Set[str],
+                        connections_to_randomize: List[ConnectionData], random: Random):
+    randomized_connections_already_shuffled = {connection: randomized_connections[connection]
+                                               for connection in randomized_connections
+                                               if connection != randomized_connections[connection]}
+    unreachable_regions_names_leading_somewhere = tuple([region for region in unreachable_regions
+                                                   if len(regions_by_name[region].exits) > 0])
+    unreachable_regions_leading_somewhere = [regions_by_name[region_name] for region_name in unreachable_regions_names_leading_somewhere]
+    unreachable_regions_exits_names = [exit_name for region in unreachable_regions_leading_somewhere for exit_name in region.exits]
+    unreachable_connections = [connections_by_name[exit_name] for exit_name in unreachable_regions_exits_names]
+    unreachable_connections_that_can_be_randomized = [connection for connection in unreachable_connections if connection in connections_to_randomize]
+
+    chosen_unreachable_entrance = random.choice(unreachable_connections_that_can_be_randomized)
+
+    chosen_reachable_entrance = None
+    while chosen_reachable_entrance is None or chosen_reachable_entrance not in randomized_connections_already_shuffled:
+        chosen_reachable_region_name = random.choice(tuple(reachable_regions))
```

Comment:
```suggestion
        chosen_reachable_region_name = random.choice(sorted(reachable_regions))
```
choice from tupled set is not deterministic.